src/NServiceBus.Transport.Msmq/MessagePump.cs

```diff
@@ -30,15 +30,19 @@ public void Dispose()
         {
             peekCircuitBreaker?.Dispose();
             receiveCircuitBreaker?.Dispose();
-            cancellationTokenSource?.Dispose();
+            messagePumpCancellationTokenSource?.Dispose();
         }
 
         public Task Initialize(PushRuntimeSettings limitations, OnMessage onMessage, OnError onError, CancellationToken cancellationToken)
         {
+            messagePumpCancellationTokenSource = new CancellationTokenSource();
+
+            var tokenForCriticalErrorAction = messageProcessingCancellationTokenSource.Token; // Prevent ObjectDisposed after endpoint shut down by using a local variable
```

Comment:
`messageProcessingCancellationTokenSource` seems to be assigned here: https://github.com/Particular/NServiceBus.Transport.Msmq/pull/225/files#diff-159e576e4eaa91dfa7405b89fc706862e8655e7c7c84ccc4aee0d5bdc8346098R85 which is in `StartReceive`?

Reply:
Good catch, assignment moved the the correct place